packages/lab/src/stepped-tracker/SteppedTracker.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+  ComponentPropsWithoutRef,
+  forwardRef,
+  ReactElement,
+  ReactNode,
+  Children,
+  useState,
+  cloneElement,
+  isValidElement,
+  useEffect,
+} from "react";
+import { clsx } from "clsx";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import { makePrefixer } from "@salt-ds/core";
+import { TrackerStep } from "./TrackerStep";
+
+import steppedTrackerCss from "./SteppedTracker.css";
+
+import useDetectTruncatedText from "./useDetectTruncatedText";
+
+const withBaseName = makePrefixer("saltSteppedTracker");
+
+export interface SteppedTrackerProps extends ComponentPropsWithoutRef<"ul"> {
+  /**
+   * The index of the current activeStep
+   */
+  activeStep: number;
+  /**
+   * Should be one or more <TrackerStep> components
+   */
+  children: ReactNode;
+}
+
+const useCheckInvalidChildren = (children: ReactNode) => {
+  useEffect(() => {
+    if (process.env.NODE_ENV !== "production") {
+      let hasInvalidChild = false;
+      Children.forEach(children, (child) => {
+        if (!isValidElement(child) || child.type !== TrackerStep) {
+          hasInvalidChild = true;
+        }
+      });
+
+      if (hasInvalidChild) {
+        console.error(
+          "Invalid children: children of <SteppedTracker> must be a <TrackerStep> component"
+        );
+      }
+    }
+  }, [children]);
+};
+
+export const SteppedTracker = forwardRef<HTMLUListElement, SteppedTrackerProps>(
+  function SteppedTracker(
+    { children, className, activeStep, ...restProps },
+    ref?
+  ): ReactElement<SteppedTrackerProps> {
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-stepped-tracker",
+      css: steppedTrackerCss,
+      window: targetWindow,
+    });
+
+    useCheckInvalidChildren(children);
+
+    const [hasTooltips, setHasTooltips] = useState(false);
+
+    // A factory function used to get a callback ref for checking truncation.
+    const getOverflowRef = useDetectTruncatedText(setHasTooltips);
+
+    const totalSteps = Children.count(children);
+
+    return (
+      <ul {...restProps} className={clsx(withBaseName(), className)} ref={ref}>
+        {Children.map(children, (child, i) => {
```

Comment:
I've used `Children.map` and `cloneElement` to inject the index for each child. As discussed recently we could do this in other ways. @joshwooding would be good to get your input

Reply:
Another way I considered was using Children.map and a context e.g.
```jsx
{Children.map(children, (child, i) => {
  <TrackerStepContext
    activeStep: activeStep,
    overflowRef: getOverflowRef(i),
    hasTooltip: hasTooltips,
    totalSteps: totalSteps,
    stepNumber: i,
  />
    {child}
  <TrackerStepContext>
}
```

Reply:
I think I prefer context in this case mainly because of the underscore props.

Reply:
I've updated to use context, let me know what you think